lib/serializer.js

```diff
@@ -144,7 +144,14 @@ function parseRelations (data, relations, options) {
     var fk = data[fkName];
 
     var fromType = utils.pluralForModel(relation.modelFrom);
-    var toType = utils.pluralForModel(relation.modelTo);
+    var toType = '';
+    if (relation.polymorphic) {
+      var discriminator = relation.polymorphic.discriminator;
+      var model = options.app.models[data[discriminator]];
+      toType = utils.pluralForModel(model);
+    } else {
+      toType = utils.pluralForModel(relation.modelTo);
+    }
```

Comment:
The reason we have been hiding foreign keys generally is that the idea with jsonapi is that these sorts of server implementation details should not be exposed to clients. 

It probably be nice to make this configurable at some point. I'm happy for you to leave it as is for now and I'll do some work at some point to clean up/improve hiding and showing foreign keys.